### PR TITLE
Watchlist caching

### LINKED ISSUE #99: Watchlist caching
As a user, I want to be able to use the watchlist without fear of hitting the API free limit

### PR DESCRIPTION
**Pull request by wwweert123**
fixes #99

### CODE REVIEW

**Review by wwweert123 — COMMENTED** [review]
💯

File: src/main/java/seedu/financialplanner/investments/WatchList.java [hunk 1]
```diff
@@ -51,17 +53,34 @@ public static WatchList getInstance() {
...
+        for (Map.Entry<String, Stock> set: stocks.entrySet()) {
+            if (set.getValue().getLastFetched() + fivemin < currentTime) {
+                queryStocks.append(set.getKey());
+                queryStocks.append(",");
+            }
+        }
+        return queryStocks;
+    }
+
+    public void fetchFMPStockPrices(StringBuilder queryStocks) throws FinancialPlannerException {
```
Comment (ryan1604) [inline src/main/java/seedu/financialplanner/investments/WatchList.java:74]: Code looks cleaner. Good job!

File: src/main/java/seedu/financialplanner/investments/WatchList.java [hunk 2]
```diff
@@ -154,6 +139,37 @@ public void extractStockInfoFromJSON(JSONObject stock, Stock stockLocal) {
...
+        String timestamp = stock.get("timestamp").toString();
+        long lastUpdated = Long.parseLong(timestamp) * 1000;
+        stockLocal.setLastUpdated(new Date(lastUpdated));
+    }
+
+    public String addStock(String stockCode) throws FinancialPlannerException {
+        if (stocks.size() >= 5) {
+            throw new FinancialPlannerException("Watchlist is full (max 5). Delete a stock to add a new one");
+        }
+        if (stocks.containsKey(stockCode.toUpperCase())) { // should already be uppercase
```
Comment (ryan1604) [inline src/main/java/seedu/financialplanner/investments/WatchList.java:152]: Do we still need the toUpperCase method if its alr in uppercase

**Review by ryan1604 — APPROVED** [review]
LGTM

File: src/main/java/seedu/financialplanner/investments/Stock.java [hunk 3]
```diff
@@ -125,4 +124,20 @@ public String getDayLow() {
     public void setDayLow(String dayLow) {
         this.dayLow = dayLow;
     }
+
+    public Date getLastUpdated() {
```
Comment (NeoMinWei) [inline src/main/java/seedu/financialplanner/investments/Stock.java:128]: good use of getters and setters

**Review by NeoMinWei — APPROVED** [review]
LGTM

File: src/main/java/seedu/financialplanner/investments/WatchList.java (hunk 2 above)
Comment (wwweert123) [reply src/main/java/seedu/financialplanner/investments/WatchList.java:152]: The saved file might be meddled with

**Pull request merged**